tests/integration_test.rs

```diff
@@ -678,6 +679,106 @@ fn test_vm_generates_frames_and_spends_gas() {
+        9999,
+        Address::from_low_u64_be(1234),
+        Address::default(),
+    );
+    let vm = VMStateBuilder::new().with_contexts(vec![context]).build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from(1234));
+}
+
+#[test]
+fn test_context_caller() {
+    let bin_path = make_bin_path_asm("context_caller");
+    let program = program_from_file(&bin_path);
+    let context = Context::new(
+        program,
+        9999,
+        Address::default(),
+        Address::from_low_u64_be(4321),
+    );
+    let vm = VMStateBuilder::new().with_contexts(vec![context]).build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from(4321));
+}
+
+#[test]
+fn test_context_code_address() {
+    let bin_path = make_bin_path_asm("context_code_address");
+    let program = program_from_file(&bin_path);
+    let context = Context::new(
+        program,
+        9999,
+        Address::from_low_u64_be(1324),
+        Address::default(),
+    );
+    let vm = VMStateBuilder::new().with_contexts(vec![context]).build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from(1324));
+}
+
+#[test]
+fn test_context_code_ergs_left() {
+    let bin_path = make_bin_path_asm("context_code_ergs_left");
+    let program = program_from_file(&bin_path);
+    let context = Context::new(program, 9999, Address::default(), Address::default());
+    let vm = VMStateBuilder::new().with_contexts(vec![context]).build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from_dec_str("9994").unwrap()); // 5 context.ergs_left
+}
+
+#[test]
+fn test_context_sp() {
+    let bin_path = make_bin_path_asm("context_sp");
+    let vm = VMStateBuilder::default().build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from_dec_str("4").unwrap());
+}
+
+#[test]
+fn test_context_get_context_u128() {
+    let bin_path = make_bin_path_asm("context_get_context_u128");
+    let vm = VMStateBuilder::default().build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from_dec_str("0").unwrap());
+}
+
+#[test]
+fn test_context_set_context_u128() {
+    // program calls set_context and then get_context, no need to pass any custom state
+    let bin_path = make_bin_path_asm("context_set_context_u128");
+    let vm = VMStateBuilder::default().build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from_dec_str("42").unwrap());
+}
+
+#[test]
+fn test_context_meta() {
+    let bin_path = make_bin_path_asm("context_meta");
+    let program = program_from_file(&bin_path);
+    let context = Context::new(program, 9999, Address::default(), Address::default());
+    let vm = VMStateBuilder::new().with_contexts(vec![context]).build();
+    let (res, _) = run_program(&bin_path, vm, &mut []);
+    assert_eq!(res, U256::from_dec_str("0").unwrap());
+}
+
```

Comment:
Maybe add a test were meta != 0, for example storing something on heap

Reply:
Done!